### PR TITLE
TabPanel rerenders items after badge changed (close #5966, T687958)

### LINKED ISSUE #5966: TabPanel repaint content when change badge for item
I paint in tab badge with total count records in datagrid.
When apply filter to grid and update badge value, content repainted and filter reset.
If enable state storing for datagrid then request to server as infinity loop.

### PR DESCRIPTION
**Pull request by churkin**

### CODE REVIEW

File: js/ui/tab_panel.js [hunk 1]
```diff
@@ -369,10 +373,14 @@ var TabPanel = MultiView.inherit({
             case "items":
```
Comment (mrjohn0011) [inline js/ui/tab_panel.js]: Try to use `_itemOptionChanged` method like it is done in the base collection widget

**Review by mrjohn0011 — CHANGES_REQUESTED** [review]
You can use an existing method to redraw only changed items

**Comment by churkin** [comment]
@lukyanovas ping

**Review by lukyanovas — APPROVED** [review]
LGTM

**Review by mrjohn0011 — APPROVED** [review]

**Review by jsdmitry — APPROVED** [review]

**Pull request merged**